src/parser.ts

```diff
@@ -245,17 +246,22 @@ function value(constant: boolean): ast.ValueNode {
   };
 }
 
-function arguments_(constant: boolean): ast.ArgumentNode[] | undefined {
+function arguments_(
+  constant: boolean,
+  fragmentArgument?: boolean
+): ast.ArgumentNode[] | ast.FragmentArgumentNode[] | undefined {
```

Comment:
nit: let's make this argument non-optional if it's not exposed? Could simply use an overload here

```suggestion
function arguments_(
  constant: boolean,
  fragmentArgument: false
): ast.ArgumentNode[] | undefined;
function arguments_(
  constant: boolean,
  fragmentArgument: true
): ast.FragmentArgumentNode[] | undefined;
function arguments_(
  constant: boolean,
  fragmentArgument: boolean
): ast.ArgumentNode[] | ast.FragmentArgumentNode[] | undefined {
```